jsonschema2pojo-core/src/main/java/org/jsonschema2pojo/Jackson2Annotator.java

```diff
@@ -129,9 +136,13 @@ public void objectBuilder(JDefinedClass clazz, JDefinedClass builder) {
                         constructor.param(field.type(), field.name())
                                 .annotate(annotation.getAnnotationClass())
                                 .param("value", str);
-                        body.assign(JExpr._this().ref(field.name()),
-                                // TODO(micah): this should respect immutability config
-                                immutableCopy(assignment, field.type()));
+
+                        if (this.getGenerationConfig().isImmutable()) {
+                            body.assign(JExpr._this().ref(field.name()),
+                                    immutableCopy(assignment, field.type()));
+                        } else {
+                            body.assign(JExpr._this().ref(field.name()), assignment);
```

Comment:
Minor style nit: maybe I'm just sleep-deprived, but I think written this way, it's slightly messy to see that the only thing that differs between the two cases is in the right hand of the assignment. Would you object to untangling those to emphasize that the assignment step and its target are the same?

```
JExpression rhs = this.getGenerationConfig().isImmutable() ?
    immutableCopy(assignment, field.type()) :
    assignment;
body.assign(JExpr._this().ref(field.name()), rhs);
```